### ISSUE TITLE
Private updates: automatically make the update public after a certain period

### ISSUE THREAD

**Issue by Betree** [opencollective/opencollective#2058]
Following on https://github.com/opencollective/opencollective/issues/1923#issuecomment-483788803
This will allow collectives to give their latest news to their backers before everyone else while still making sure the update becomes publicly visible after a certain period of time.

# Specs

## Frontend

- [ ] Add a date input on edit/create update, left empty by default
  - Field label: `Automatically make the update public on this date:`

## API

- [ ] Add a `makePublicOn` field to updates (type=Date, default=null)
- [ ] Add a CRON that runs a query every day
  - For all updates with `"makePublicOn" <= NOW()`, set `isPrivate` to false
- [ ] Add a unit test that proves the functionality 

---

**General guidelines**
- The strings must be internationalized. See [/help/developers/translations](https://docs.opencollective.com/help/developers/translations)
- Whenever it's possible we must use `styled-components` to write styles. See [OC Styleguide](https://opencollective-styleguide.now.sh/)
- Icons must be imported from the [styled-icons](http://styled-icons.js.org/) library.
- Tests written with Cypress must follow our [good practices](https://docs.opencollective.com/help/developers/testing-with-cypress) conventions.

**Comment by Betree** [comment]
@josegonzalez are you still willing to put a $100 bounty on this one?

**Comment by josegonzalez** [comment]
Yes, I am :)

**Comment by Betree** [comment]
:gift: :money_with_wings: A **$100** bounty from @josegonzalez @ [Dokku](https://opencollective.com/dokku) has been attached to this issue. 
:hourglass_flowing_sand: Time span to complete the issue once assigned: 2 weeks

---

Anyone submitting a Pull Request will be rewarded with $100 when the Pull Request is reviewed, accepted and merged.

---

:arrow_right: [More info on bounties](https://github.com/opencollective/opencollective-frontend/edit/master/BOUNTY.md)

**Comment by brymut** [comment]
Hi @Betree @josegonzalez, more than happy to have a look at this while I wait for a couple of pull-requests' reviews.

**Comment by brymut** [comment]
Hi @Betree, currently moving onto the front-end side of the specs for this issue. I was wondering if there are any mockups for how the date input should look like on Figma or something. I've come across this mock-up shared recently that has the private update checkbox, but I haven't seen anything for the date input yet.  Could you please point me toward what I can use, or should I just start and wait for the exact design spec and follow that after I've got something going?

**Comment by Betree** [comment]
@brymut We don't have a calendar component yet, but `<StyledInput type="date" />` should do the job. It will show the native browser date picker input with some basic styling (border-radius, color...etc). Not the best looking but good for accessibility :)

**Comment by brymut** [comment]
Hi @Betree , have been working on this and noticed that there is a calendar input used when submitting a new expense https://opencollective.com/engineering/expenses/new 

https://github.com/opencollective/opencollective-frontend/blob/d95d96430ffcaf15f36728c8aaaa1cfcf2586bda/src/apps/expenses/components/CreateExpenseForm.js#L412

It uses a `<InputField
                    defaultValue={new Date()}
                    type="date"
                    ...`   might that be a way to go? Or should I just stick to the `<StyledInput type="date" />`?

**Comment by brymut** [comment]
Hi again @Betree. I've created a pull request for the work I've started on this, however, I feel like I am currently blocked on completing the front-end spec on this while I wait on opencollective/opencollective-frontend#1796 to get merged.

**Comment by Betree** [comment]
Hi @brymut, sorry for the delay.

I'd prefer not to, the dependencies behind this component are quite heavy and it actually doesn't really look better than the default browser's calendar:

![image](https://user-images.githubusercontent.com/1556356/58974917-f593fe00-87c3-11e9-9f12-fc3c006ef07c.png)

Going native ensure that we keep maximum accessibility and localization. Unless there's a good reason for using `<InputField />` rather than `<StyledInput />` (which is just a regular `input` with some styling on it) we shouldn't use it.

**Comment by Betree** [comment]
> Hi again @Betree. I've created a pull request for the work I've started on this, however, I feel like I am currently blocked on completing the front-end spec on this while I wait on [opencollective/opencollective-frontend#1796](https://github.com/opencollective/opencollective-frontend/pull/1796) to get merged.

@brymut We should wait for opencollective/opencollective-frontend#1796 to be merged, and ideally for https://github.com/opencollective/opencollective/issues/2090 to be completed to avoid any conflict.

Let's put this one on hold while we wait for this to be ready. We'll give you the time that you need once it is to ensure the bounty stays assigned to you.

**Comment by flickz** [comment]
@brymut & @Betree I think you're good to go here.

**Comment by Betree** [comment]
Alright, thanks @flickz :slightly_smiling_face: 

@brymut ready to work on this?

**Comment by brymut** [comment]
Yep @Betree, will get back on this over the weekend. Thanks @flickz!

**Comment by Betree** [comment]
@brymut any news on that one?

**Comment by Betree** [comment]
De-assigning @brymut because of lack of activity

**Comment by leonardohof** [comment]
@Betree Can assign to me?

**Comment by flickz** [comment]
@Betree Might be busy, we're having an offsite retreat, feel free to take it on. I'll assign it to you

**Comment by leonardohof** [comment]
@flickz thanks!

**Comment by Betree** [comment]
:trophy: This issue is now completed by @leonardohof 

To claim the **$300** bounty, you can either:
- Submit an invoice to https://opencollective.com/engineering/expenses/new (please mention the issue number in the expense's title)
- Or ask for an Open Collective gift card of the same value (to donate to other collectives)
